Marlin/src/pins/pins_GTM32_PRO_VB.h

```diff
@@ -64,66 +79,77 @@
+// NOTE: Numbering here is made according to EXT connector numbers,
+//       the FANx_PWM line numbering in the schematics is reverse.
+//       That is, E0_*_PIN are the E2_* lines connected to E2_A1 step
+//       stick that drives the EXT0 output on the board.
+//
+#define E0_STEP_PIN        PC14
+#define E0_DIR_PIN         PC13
+#define E0_ENABLE_PIN      PC15
 
 #define E1_STEP_PIN        PA0
 #define E1_DIR_PIN         PB6
 #define E1_ENABLE_PIN      PA1
 
-#define E2_STEP_PIN        PC14
-#define E2_DIR_PIN         PC13
-#define E2_ENABLE_PIN      PC15
+#define E2_STEP_PIN        PB2
+#define E2_DIR_PIN         PB11
+#define E2_ENABLE_PIN      PC4
 
 //
 // Heaters / Fans
 //
-#define HEATER_0_PIN       PB0
-#define HEATER_1_PIN       PB5
-#define HEATER_2_PIN       PB4
-#define HEATER_BED_PIN     PB1
+#define HEATER_0_PIN       PB0   // EXT0 port
+#define HEATER_1_PIN       PB5   // EXT1 port
+#define HEATER_2_PIN       PB4   // EXT2 port
+#define HEATER_BED_PIN     PB1   // CON2X3 hotbed port
 
-#define FAN_PIN            PB7   // TODO: Add functionality
-#define FAN1_PIN           PB8   // TODO: Add functionality
-#define FAN2_PIN           PB9   // TODO: Add functionality
+//
+// These are FAN PWM pins on EXT0..EXT2 connectors.
+//
+//#define FAN_PIN            PB9   // EXT0 port
+#define ORIG_E0_AUTO_FAN_PIN PB9   // EXT0 port, used as main extruder fan
+#define FAN1_PIN           PB8   // EXT1 port
+#define FAN2_PIN           PB7   // EXT2 port
 
 //
 // Temperature Sensors
 //
-#define TEMP_0_PIN         PC2   // PORT NUMBERING
-#define TEMP_1_PIN         PC1   // PORT NUMBERING
-#define TEMP_2_PIN         PC0   // PORT NUMBERING
-#define TEMP_BED_PIN       PC3   // PORT NUMBERING
+#define TEMP_0_PIN         PC2   // EXT0 port
+#define TEMP_1_PIN         PC1   // EXT1 port
+#define TEMP_2_PIN         PC0   // EXT2 port
+#define TEMP_BED_PIN       PC3   // CON2X3 hotbed port
 
 //
 // Misc. Functions
 //
 #define LED_PWM            PD12   // External LED, pin 2 on LED labeled connector
-//#define RESET_PIN        NRST
-#define T_PEN              PE6
-
-#define BEEPER_PIN         PB10
 
 //
 // LCD / Controller
 //
 #if HAS_SPI_LCD
 
   #if ENABLED(REPRAP_DISCOUNT_SMART_CONTROLLER)
+    //
+    // LCD display on J2 FFC40
+    // Geeetech's LCD2004A Control Panel is very much like
+    // RepRapDiscount Smart Controller, but adds an FFC40 connector
+    //
     #define LCD_PINS_RS    PE6   // CS chip select /SS chip slave select
     #define LCD_PINS_ENABLE PE14 // SID (MOSI)
     #define LCD_PINS_D4    PD8   // SCK (CLK) clock
     #define LCD_PINS_D5    PD9
     #define LCD_PINS_D6    PD10
     #define LCD_PINS_D7    PE15
 
-    #define BTN_EN1        PE8
-    #define BTN_EN2        PE9
-    #define BTN_ENC        PE13
-
-    #define SD_DETECT_PIN  PC7
-    #define KILL_PIN       -1
+  #else
+    //
+    // Serial LCDs can be implemented in ExtUI
+    //
+    //#define LCD_UART_TX    PD8
+    //#define LCD_UART_RX    PD9
```

Comment:
If the protocols are well-documented, the LCD can be supported through ExtUI following the included DGUS or MalyanLCD serial touch screen examples.